go-controller/pkg/ovn/egressip_test.go

```diff
@@ -3714,6 +3755,227 @@ var _ = ginkgo.Describe("OVN master EgressIP Operations", func() {
+								"router-port": types.GWRouterToExtSwitchPrefix + "GR_" + node1Name,
+							},
+						},
+						&nbdb.LogicalSwitchPort{
+							UUID: types.EXTSwitchToGWRouterPrefix + types.GWRouterPrefix + node2Name + "UUID",
+							Name: types.EXTSwitchToGWRouterPrefix + types.GWRouterPrefix + node2Name,
+							Type: "router",
+							Options: map[string]string{
+								"router-port": types.GWRouterToExtSwitchPrefix + "GR_" + node2Name,
+							},
+						},
+					},
+				})
+				gomega.Expect(fakeOvn.controller.WatchEgressNodes()).To(gomega.Succeed())
+				gomega.Eventually(getEgressIPAllocatorSizeSafely).Should(gomega.Equal(0))
+
+				_, ip1V6Sub, err := net.ParseCIDR(node1IPv6)
+				_, ip2V4Sub, err := net.ParseCIDR(node2IPv4)
+
+				_, err = fakeOvn.fakeClient.KubeClient.CoreV1().Nodes().Create(context.TODO(), &node1, metav1.CreateOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Eventually(getEgressIPAllocatorSizeSafely).Should(gomega.Equal(1))
+				gomega.Expect(fakeOvn.controller.eIPC.allocator.cache).To(gomega.HaveKey(node1.Name))
+				gomega.Expect(fakeOvn.controller.eIPC.allocator.cache[node1.Name].egressIPConfig.V6.Net).To(gomega.Equal(ip1V6Sub))
+
+				_, err = fakeOvn.fakeClient.KubeClient.CoreV1().Nodes().Create(context.TODO(), &node2, metav1.CreateOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Eventually(getEgressIPAllocatorSizeSafely).Should(gomega.Equal(2))
+				gomega.Eventually(isEgressAssignableNode(node1.Name)).Should(gomega.BeTrue())
+				gomega.Eventually(isEgressAssignableNode(node2.Name)).Should(gomega.BeTrue())
+				gomega.Expect(fakeOvn.controller.eIPC.allocator.cache).To(gomega.HaveKey(node1.Name))
+				gomega.Expect(fakeOvn.controller.eIPC.allocator.cache).To(gomega.HaveKey(node2.Name))
+
+				cachedEgressNode1 := fakeOvn.controller.eIPC.allocator.cache[node1.Name]
+				cachedEgressNode2 := fakeOvn.controller.eIPC.allocator.cache[node2.Name]
+				gomega.Expect(cachedEgressNode1.egressIPConfig.V6.Net).To(gomega.Equal(ip1V6Sub))
+				gomega.Expect(cachedEgressNode2.egressIPConfig.V4.Net).To(gomega.Equal(ip2V4Sub))
+
+				// Explicitly call check reachibility so we need not to wait for slow periodic timer
+				checkEgressNodesReachabilityIterate(fakeOvn.controller)
+				gomega.Expect(cachedEgressNode1.isReachable).To(gomega.BeTrue())
+				gomega.Expect(cachedEgressNode2.isReachable).To(gomega.BeTrue())
+
+				tests := []struct {
+					desc            string
+					node1FailProbes bool
+					node2FailProbes bool
+					init            func(hcc1, hcc2 *fakeEgressIPHealthClient)
+				}{
+					{
+						desc:            "disconnect nodes",
+						node1FailProbes: true,
+						node2FailProbes: true,
+						init: func(hcc1, hcc2 *fakeEgressIPHealthClient) {
+							hcc1.Disconnect()
+							hcc2.Disconnect()
+						},
+					},
+					{
+						desc:            "connect node1",
+						node2FailProbes: true,
+					},
+					{
+						desc: "node1 connected, connect node2",
```

Comment:
how is this test case different from desc: "node1 and node2 connected", they both have the same settings?

Reply:
good point. I think this is not a useful test case... I am going to remove it to make that less confusing.